Lib/subprocess.py

```diff
@@ -773,6 +778,11 @@ def __init__(self, args, bufsize=-1, executable=None,
         if not isinstance(bufsize, int):
             raise TypeError("bufsize must be an integer")
 
+        if pipesize is None:
```

Comment:
I suggest not even allowing None, just let someone passing None trigger the TypeError below.

Reply:
[This is also done for bufsize](https://github.com/python/cpython/blob/58925f3bdeaa904b64e8e5257c594c1268c9de47/Lib/subprocess.py#L776), so this choice was made to be consistent with the rest of the code base.

Furthermore I would argue that `some_arg=None` is the Pythonic way of letting a function use the default. `some_arg=-1` feels more C-like. But of course this is opinionated.